### PR TITLE
fix(AlgoliaSearch): Add support for keyboard navigation

### PR DESCRIPTION
**Pull request by kotva006**
#### Summary of changes
Adding the `getItemUrl()` function enables keyboard navigation for the autocomplete library.
Allowing users to use the arrow and enter keys to interact with the search results.

**Asana Ticket:** [Search Bar Not Working on the top of the navigation](https://app.asana.com/0/home/1201123880594717/1206592383985332)

---

#### General checks
* [ ] **Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?** This is a good practice that can facilitate easier reviews.
* [ ] **Testing.** Do the changes include relevant passing updates to tests? This includes updating screenshots. Preferably tests are run locally to verify that there are no test failures created by these changes, before opening a PR.
* [ ] **Tech debt.** Have you checked for tech debt you can address in the area you're working in? This can be a good time to address small issues, or create Asana tickets for larger issues.

#### New UI, or substantial UI changes
* [ ] **Cross-browser compatibility** is less of an issue now that we're no longer supporting IE, but changes still need to work as expected in Safari, Chrome, and Firefox.
* [ ] **Are interactive elements accessible?** This includes at minimum having [relevant keyboard interactions](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#keyboard-interaction) and [visible focus](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus), but can also include verification with screen reader testing.
* [ ] **Other accessibility checks** such as sufficient [color constrast](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#color-contrast), or whether the layout holds up at 200% [zoom level](https://thinkdobecreate.com/articles/4-required-tests-before-shipping-new-features/#visible-focus).

#### New endpoints, or non-trivial changes to current endpoints
* [ ] **Have we load-tested** any new pages or internal API endpoints that will receive significant traffic? [See load testing docs](./../load_tests/README.md)
* [ ] **If this change involves routes,** does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?

### CODE REVIEW

File: assets/ts/ui/autocomplete/plugins/algolia.ts [hunk 1]
```diff
@@ -23,6 +23,10 @@ export default function createAlgoliaBackendPlugin(
             templates: {
               item: AlgoliaItemTemplate
             },
+            getItemUrl({ item }) {
+              const { url, _content_url } = item as AutocompleteItem;
```
Comment (anthonyshull) [inline assets/ts/ui/autocomplete/plugins/algolia.ts:27]: The underscore signifies that this property is private. We should change it if it's meant to be public.

**Review by anthonyshull — CHANGES_REQUESTED** [review]
We shouldn't use a private property.

File: assets/ts/ui/autocomplete/plugins/algolia.ts (hunk 1 above)
Comment (kotva006) [reply assets/ts/ui/autocomplete/plugins/algolia.ts:27]: We use it in other places [as added at this location](https://github.com/mbta/dotcom/blob/master/assets/ts/ui/autocomplete/templates/algolia.tsx#L27) so I think it is fair to keep using it in this location.

Comment (anthonyshull) [reply assets/ts/ui/autocomplete/plugins/algolia.ts:27]: Or, we could take the time to fix it...up to you.

**Review by anthonyshull — APPROVED** [review]

**Pull request merged**